adminapi/tests/__init__.py

```diff
@@ -810,3 +827,53 @@ def test_API_no_matching_model_name_from_url_error_put(self):
                 "detail": "Model or App does not exist"
             }
         )
+
+
+class MultiPartFormFieldTest(TestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+
+    @classmethod
+    def setUpClass(cls):
+        cls.image_model = ImageModel()
+        cls.image_model.title = "Image model 1"
+        cls.image_model.save()
+
+    @classmethod
+    def tearDownClass(cls):
+        super(MultiPartFormFieldTest, cls).tearDownClass()
+
+    def test_image_upload_API(self):
+        client = APIClient()
+        response = client.post(
+            reverse("test-generic-list", args=["adminapi", "imagemodel"]),
+            {
+                "title": "Image model 2",
+                "image": open(IMAGE_PATH)
+            },
+            format="multipart"
+        )
+        self.assertEqual(response.status_code, 201)
+        # Test that image was saved to directory and then delete it
+        self.assertTrue(os.path.isfile(glob.glob(RES_DIR+"/image_*.jpg")[0]))
+        os.remove(glob.glob(RES_DIR+"/image_*.jpg")[0])
+        parsed_json = json.loads(response.content)
+        self.assertEqual(parsed_json["title"], "Image model 2")
+        self.assertEqual(parsed_json["id"], 2)
+
+    def test_image_update_API(self):
+        client = APIClient()
+        response = client.put(
+            reverse("test-generic-detail", args=["adminapi", "imagemodel", 1]),
+            {
+                "image": open(IMAGE_PATH)
+            },
+            format="multipart"
```

Comment:
format="multipart" is compatible with Django's existing RequestFactory. Would need to set TEST_REQUEST_RENDERER_CLASSES in test_settings.py to support more format types.

To use the traditional content_type argument data would need to be explicitly encoded. For multipart/form-data this includes setting the boundary data and adding it to the header.